src/build/app/settings.rs

```diff
@@ -688,7 +688,28 @@ pub enum AppSettings {
     /// ```
     /// [``]: ./struct..html
     /// [`App`]: ./struct.App.html
-    DisableVersion,
+    DisableVersionFlag,
+
+    /// Disables `-V` and `--version` for all [`subcommand`]s of this [`App`]
+    /// (Defaults to `false`; subcommands *do* have version flags.)
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # use clap::{App, AppSettings, ErrorKind};
+    /// let res = App::new("myprog")
+    ///     .version("v1.1")
+    ///     .setting(AppSettings::DisableVersionForSubcommands)
+    ///     .subcommand(App::new("test"))
+    ///     .try_get_matches_from(vec![
+    ///         "myprog", "test", "-V"
+    ///     ]);
+    /// assert!(res.is_err());
+    /// assert_eq!(res.unwrap_err().kind, ErrorKind::UnknownArgument);
+    /// ```
+    /// [`App`]: ./struct.App.html
+    /// [``]: ./struct..html
+    DisableVersionForSubcommands,
```

Comment:
Yup, if we introduce `version` subcommand in the future, this setting will disable both (because it is more related to subcommands behaviour than arg behaviour), but `DisableVersionFlag` will only disable the flag and we will introduce a new `DisableVersionSubcommand` to disable the subcommand.